### PR TITLE
chore(cargo): Update dependency

### PR DESCRIPTION
**Pull request by dariusc93**
### What this PR does 📖

- Generate thumbnails for images internally rather than in uplink
- Remove function generating thumbnail for image
- Added small helper to generate base64 based on internal file type

### Which issue(s) this PR fixes 🔨

- N/A

### Special notes for reviewers 🗒️
- Not every image will generate a thumbnail. At the time, only PNG, JPG/JPEG, GIF, ICO, BMP, and WEBP will be supported for generating a thumbnail
- Clicking on an thumbnail in the file explorer will show the image/thumbnail at 500x500 (default, but configurable though recommend to leave it at this size or lower it down). A separate ticket may need to be opened to grab the image itself to display it in the modal rather than the thumbnail

### Additional comments 🎤
- Due to changes to `File` in warp, this will not be compatible with previous index (does not require new identity).
- Should unblock #728

### CODE REVIEW

**Comment by stavares843** [comment]
**AI CODE REVIEW**


General feedback:

The code changes seem to be focused on generating thumbnails for images internally instead of using the "uplink" functionality.
A function that generated thumbnails for images has been removed.
A small helper has been added to generate base64 data based on the internal file type.
The pull request does not mention any specific issues that it fixes.
Comments and considerations:

It is mentioned that only certain image formats (PNG, JPG/JPEG, GIF, ICO, BMP, and WEBP) will be supported for generating thumbnails.
Clicking on a thumbnail in the file explorer will display the image/thumbnail at 500x500 size, which is the default but can be configured.
There is a suggestion to open a separate ticket to display the full image in the modal instead of the thumbnail.
There is a note about compatibility issues with the previous index due to changes in the File module in Warp.
Overall, the changes seem to be aimed at improving thumbnail generation for images. It would be helpful to have access to the actual code changes in order to provide more specific feedback.

**Review by stavares843 — COMMENTED** [review]
Captura de ecrã 2023-05-17, às 13 21 21

the jpg one didn't generated a thumb

**Review by stavares843 — COMMENTED** [review]
Captura de ecrã 2023-05-17, às 13 21 47


non image files shows this thumb

**Review by stavares843 — COMMENTED** [review]
gif shows a thumb but isn't animated

Captura de ecrã 2023-05-17, às 13 23 00

**Review by stavares843 — COMMENTED** [review]
Captura de ecrã 2023-05-17, às 13 23 50


ico file no thumb displayed

**Review by stavares843 — COMMENTED** [review]
Captura de ecrã 2023-05-17, às 13 25 07


WEBP no thumb

**Comment by stavares843** [comment]
minor feedback above, all in macOS, m1

**Review by stavares843 — APPROVED** [review]

**Comment by dariusc93** [comment]
> Captura de ecrã 2023-05-17, às 13 21 21
> 
> the jpg one didn't generated a thumb

Could you provide the jpg so I could check. Every jpg ive used produced a thumbnail

> Captura de ecrã 2023-05-17, às 13 21 47
> 
> non image files shows this thumb

It should be showing up with its default images in uplink for anything that dont generate a thumbnail. I can double check that here shortly

> gif shows a thumb but isn't animated
> Captura de ecrã 2023-05-17, às 13 23 00

That is expected since its producing a thumbnail and clicking on the image only show the thumbnail itself and not the direct contents (referring back comment mentioning of wiring it to fetch the image for the preview)

> Captura de ecrã 2023-05-17, às 13 23 50
> 
> ico file no thumb displayed


> Captura de ecrã 2023-05-17, às 13 25 07
> 
> WEBP no thumb

Ill check the ICO (havent tested this yet, but assumed it will work) and try with more samples of webp. Only used one image but never tried any other so it might be related to how the library being used is detecting it up.

**Comment by dariusc93** [comment]
Making one minor change to add a function to determine thumbnail format than to depend on the file format itself

File: common/src/warp_runner/mod.rs [hunk 1]
```diff
@@ -338,10 +338,15 @@ async fn warp_initialization(tesseract: Tesseract) -> Result<manager::Warp, warp
...
         .map(|mp| Box::new(mp) as Account)?;
 
-    let storage =
-        warp_fs_ipfs::IpfsFileSystem::new(account.clone(), Some(FsIpfsConfig::production(path)))
-            .await
-            .map(|ct| Box::new(ct) as Storage)?;
+    let mut config = FsIpfsConfig::production(path);
+    config.thumbnail_size = (500, 500);
+
+    //Note: If this is set to true, uncomment logic in `thumbnail_to_base64`
```
Comment (dariusc93) [inline common/src/warp_runner/mod.rs]:
```suggestion
```

**Review by WanderingHogan — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Pull request merged**